docs/rails.md

```diff
@@ -77,6 +77,28 @@ class PostsController < ApplicationController
 end
 ```
 
+### Callbacks
+
+You can add and skip callbacks which are called before, after and around authorization:
+
+```ruby
+class PostsController < ApplicationController
+  before_action :authorize!
+  before_authorize :set_current_user, only: :index
```

Comment:
Thanks for the suggestion and the PR, but I have to reject.

And your example demonstrates the main reason: adding yet-another callback introduces a new indirection layer. I don't want to encourage that kind of code design.

Another reason is that there are at least two ways to solve the problem, which, IMO, are easier to work with:

```ruby
# first, we can define the callback with all the dependencies like this
before_action do
  set_current_user
  authorize!
end

# secondly, we can use memoization for `current_user` and avoid the `set_current_user` callback altogether
def current_user
  @current_user ||= User.find(params[:id])
end
```

Reply:
Sorry to hear that.

I agree that the example you give here is clearer, but it doesn't work for us; the `load_current_context` and `authorize` callbacks are defined in our `ApplicationController`, but our third callback which depends on `load_current_context` is in a descendant of `ApplicationController`.

Anyway, thank you for looking at my PR.

Reply:
So, you're kinda trying to inject a callback in between `load_current_context` and authorization?

I think, what could be useful is to make `authorize!` to always be called in the end, after all other `before_action` callbacks. I guess, you can achieve this by using an `around_action` callback:

```ruby
around_action do
  authorize!
  yield
end
```

Reply:
Thanks for the suggestion.

What we've gone with in the end is to skip our default `authorize!` callback in the child controller, and then to call it again after we load our dependency:

```ruby
class ApplicationController
  before_action :load_current_context
  before_action :authorize!

  # ... and a bunch of other stuff
end

class ChildController < ApplicationController
  skip_before_action :authorize!
  before_action :load_dependency, :authorize!
end
```

That way we know that `authorize!` gets called after the parent's `load_current_context` and after the child's `load_dependency`.

It's not lovely; I'd have preferred to be able to register the `load_dependency` callback in the child as happening at a particular point in the chain, either with a `before_authorize` callback as in this PR, or if there was some way to insert a `before_action` callback at a particular point.  But it works.